### ISSUE TITLE
`base` does no longer accept a tarball

### ISSUE THREAD

**Issue by mering** [bazel-contrib/rules_oci#700]
In v1, `base` was accepting a tarball. I assume this is the reason why it sets `allow_single_file = True`. In v2, this fails with the following error:
```
ls: unknown io error: '"bazel-out/k8-fastbuild/bin/path/image_tarball/tarball.tar/blobs/*/*"', 'Os { code: 20, kind: NotADirectory, message: "Not a directory" }'
```

The corresponding code is https://github.com/bazel-contrib/rules_oci/blob/c74c2d35fb65450aed690815ee6e92e1b753ed1e/oci/private/image.sh#L33-L48

Could you detect if `base` is a tarball and extract it in this case?

We use this feature to gradually migrate our large code base from `rules_docker` to `rules_oci`.

**Comment by thesayyn** [comment]
oci_image never accepted a tarball as the base, if it did though, probably because of crane we were running at the time. We won't be supporting tarballs as they are usually huge and not cache friendly.

**Comment by mering** [comment]
@thesayyn I agree that one shouldn't _create_ tarballs as outputs but it might be necessary to support tarballs as _inputs_ to support migrations _to_ rules_oci.

So this is about unpacking a tarball _which already exists_ instead of _creating a new one_. So this doesn't make things worse as they already are today but instead paves the way to a better future with fewer tarballs.

**Comment by thesayyn** [comment]
I agree with the sentiment here, however adding this feature back is not something i'd be warm to, i'd be much comfortable with an example showing use of regctl with a genrule to unpack a tarball into an oci-layout which is then used as the base.

**Comment by mering** [comment]
> I agree with the sentiment here, however adding this feature back is not something i'd be warm to, i'd be much comfortable with an example showing use of regctl with a genrule to unpack a tarball into an oci-layout which is then used as the base.

Or maybe providing an explicit `oci_import()` rule to unpack the tar. Preferably also backporting to 1.x to allow for a smoother upgrade path (using that explicit rule already in 1.x and then upgrade).

**Comment by mering** [comment]
@thesayyn which `regctl` command would you use to unpack a tarball into an oci-layout?

**Comment by thesayyn** [comment]
`regctl import path/to/tarball oci-layout://path/to/output`

**Comment by thesayyn** [comment]
Something like genrule would work but genrules no longer support outputting a directory, so you'd have to create a custom rule.

**Comment by mering** [comment]
> `regctl image import path/to/tarball oci-layout://path/to/output`

Thanks, I couldn't figure out the `oci-layout://` prefix when reading their docs.

**Comment by mering** [comment]
@thesayyn Thanks for your example!

I translated it into a rule:
```starlark
def _oci_import_impl(ctx):
    output = ctx.actions.declare_directory(ctx.label.name)
    ctx.actions.run(
        outputs = [output],
        inputs = [ctx.file.tarball],
        executable = ctx.toolchains["@rules_oci//oci:regctl_toolchain_type"].regctl_info.binary,
        arguments = [
            "image",
            "import",
            "ocidir://%s" % output.path,
            ctx.file.tarball.path,
        ],
        mnemonic = "ExtractContainerTarball",
    )
    return DefaultInfo(
        files = depset([output]),
        runfiles = ctx.runfiles(files = [output]),
    )

oci_import = rule(
    implementation = _oci_import_impl,
    doc = "Imports an image tarball into an oci-layout directory",
    attrs = {
        "tarball": attr.label(
            allow_single_file = [".tar"],
        ),
    },
    toolchains = [
        "@rules_oci//oci:regctl_toolchain_type",
    ],
)
```
